sunpy/map/mapbase.py

```diff
@@ -1891,6 +1893,77 @@ def draw_limb(self, axes=None, **kwargs):
 
         return [circ]
 
+    @u.quantity_input
+    def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = None,
+                        axes=None, top_right=None, **kwargs):
+        """
+        Draw a rectangle defined in world coordinates on the plot using Quadrangle from astropy.
+
+        This draws a rectangle that has corners at ``(bottom_left, top_right)``,
+        and has sides parallel to coordinate axes of the map.
+
+        If ``width`` and ``height`` are specified, they are respectively added to the
+        longitude and latitude of the ``bottom_left`` coordinate to calculate a
+        ``top_right`` coordinate.
+
+        Parameters
+        ----------
+        bottom_left : `~astropy.coordinates.SkyCoord`
+            The bottom-left coordinate of the rectangle. It can
+            have shape ``(2,)`` to simultaneously define ``top_right``.
+        top_right : `~astropy.coordinates.SkyCoord`
+            The top-right coordinate of the rectangle.
+        width : `astropy.units.Quantity`, optional
+            The width of the rectangle. Required if ``top_right`` is omitted.
+        height : `astropy.units.Quantity`
+            The height of the rectangle. Required if ``top_right`` is omitted.
+        axes : `matplotlib.axes.Axes`
+            The axes on which to plot the rectangle, defaults to the current
+            axes.
+
+        Returns
+        -------
+        rect : `list`
+            A list containing the `~astropy.visualization.wcsaxes.patches.Quadrangle` object, after
+            it has been added to ``axes``.
+
+        Notes
+        -----
+        If user's astropy version is < 4.2, raises an error to use draw_rectangle instead.
+
+        Extra keyword arguments to this function are passed through to the
+        `~astropy.visualization.wcsaxes.patches.Quadrangle` instance.
+        """
+        if version.parse(astropy.__version__) < version.parse("4.2.0"):
+            ImportError('Astropy >= 4.2 is required for this method, use draw_rectangle instead.')
+        else:
+            if isinstance(top_right, u.Quantity) and isinstance(width, u.Quantity):
+                # The decorator assigns the first positional arg to top_right and so on.
+                height = width
+                width = top_right
+                top_right = None
+
+            bottom_left, top_right = get_rectangle_coordinates(
+                bottom_left, top_right=top_right, width=width, height=height)
+
+            width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
+            height = Latitude(top_right.spherical.lat - bottom_left.spherical.lat)
+
+            if not axes:
+                axes = plt.gca()
+
+            coord = bottom_left.transform_to(self.coordinate_frame)
+            kwergs = {
+                "transform": wcsaxes_compat.get_world_transform(axes),
```

Comment:
```suggestion
                "transform": axes.get_transform(bottom_left),
```
I think?

Reply:
This doesn't work as of now, working on understanding the use of axes a little better. I'll make a commit regarding this soon.

Reply:
This is what you need to do:
```suggestion
                "transform": axes.get_transform(bottom_left.frame.replicate_without_data())
```